src/renderer/components/utils/SameFileImportManager.tsx

```diff
@@ -25,35 +25,51 @@ interface Props  {
     search?: any;
     searchResult?: any;
     importOpdsEntry?: any;
+    downloads?: any[];
 }
 
 class SameFileImportManager extends React.Component<Props> {
     public componentDidUpdate(oldProps: Props) {
-        const { searchResult, lastImport } = this.props;
+        const { searchResult, lastImport, downloads } = this.props;
 
         if (searchResult !== oldProps.searchResult) {
             if (searchResult.length === 0) {
-                this.props.importOpdsEntry(
-                    {
-                        url: lastImport.publication.url,
-                        base64OpdsPublication: lastImport.publication.base64OpdsPublication,
-                        downloadSample: lastImport.downloadSample,
-                        title: lastImport.publication.title,
-                    },
-                );
+                this.importOpds();
             } else {
                 this.props.displayImportDialog(lastImport.publication, lastImport.downloadSample );
             }
         }
 
         if (lastImport !== oldProps.lastImport) {
-            this.props.search({text: lastImport.publication.title});
+            const foundInCurrentDownload = downloads.findIndex(
+                (value) => {
+                    return value.publicationTitle === lastImport.publication.title;
```

Comment:
The type of `lastImport` is `ImportState`:
https://github.com/readium/readium-desktop/blob/develop/src/common/redux/states/import.ts
```
export interface ImportState {
    publication: any;
    downloadSample: boolean;
}
```

...but the type of `publication` is `any`...so by reverse-engineering the type system I was able to find this:
```
{
url: string;
base64OpdsPublication: string;
title: string;
}
```
(there may be more properties, I am not sure)

Could you please help me (and other contributors) by replacing `any` with the actual type of `publication`? Many thanks!